src/public/components/router.js

```diff
@@ -1,6 +1,75 @@
-import K, * as U from "karet.util"
-import React     from "karet"
+import * as R     from "ramda"
+import * as L     from "partial.lenses"
+import K, * as U  from "karet.util"
+import React      from "karet"
+import toRegex    from 'path-to-regexp'
 
-export default U.withContext(({routes, NotFound}, {path}) => U.fromKefir(K(path, path =>
-  React.createElement(routes[path] || NotFound)
-)))
+// TODO: Make it possible to add routes more than once?
+
+// begin routes initialization
+const expandRoutes = rawRoutes =>
+  R.pipe( R.toPairs
+        , R.map( R.zipObj( [ 'route', 'component' ] ) )
+        )( rawRoutes )
+
+const addRegexes = expandedRoutes =>
+  R.map( route => R.pipe( L.get( 'route' )
+                        , toRegex
+                        , L.set( 'regex' )
+                        )( route )( route )
+       )( expandedRoutes )
+
+const partitionDynamicAndStatic = routesWithRegexes =>
+  R.partition( R.pipe( L.get( [ 'regex', 'keys' ] )
+                     , R.length
+                     , R.lt( 0 )
+                     )
+             )( routesWithRegexes )
+
+const prepareRoutes = rawRoutes =>
+  R.pipe( expandRoutes
+        , addRegexes
+        , partitionDynamicAndStatic
+        , R.zipObj( [ 'dynamicRoutes', 'staticRoutes' ] )
+        )( rawRoutes )
+// end routes initialization
+
+//
+
+// begin route resolution
+const resolveStatic = ( path, { staticRoutes } ) =>
+  R.find( R.propEq( 'route', path ) )( staticRoutes )
+
+const prepareParams = ( keys, matches ) =>
+  R.pipe( R.map( L.get( 'name' ) )
+        , R.zipObj
+        )( keys, matches )
+
+const resolveDynamic = ( path, { dynamicRoutes } ) =>
+  R.reduce( ( acc, { component, regex } ) =>
+              R.ifElse( R.isNil
+                      , R.always( acc )
+                      , match =>
+                          R.reduced( { component
+                                     , props: prepareParams( regex.keys, R.tail( match ) )
+                                     }
+                                   )
+                      )( regex.exec( path ) )
+          , false
+          )( dynamicRoutes )
+
+const resolveRoute = ( { routes, NotFound }, { path }  ) =>
+  U.fromKefir( K( path, path =>
+                  { const { component, props } =
```

Comment:
Is there a way to get rid of this temporary `const` while maintaining readability or, hopefully, improving readability? As you can see, I try to avoid creating temporary container variables and using `return` statements in my arrow functions where I can.